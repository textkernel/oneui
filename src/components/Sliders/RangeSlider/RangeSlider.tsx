import * as React from 'react';
import Slider from 'rc-slider/lib/Slider';
import '../Slider.scss';

/** See technical notes in Slider.tsx */
const RangeSlider = ({ defaultValue = [0, 100], ...rest }) => (
    <Slider range defaultValue={defaultValue} {...rest} />
);

RangeSlider.displayName = 'RangeSlider';

export { RangeSlider };
