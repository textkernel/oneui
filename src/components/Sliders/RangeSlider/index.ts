export { RangeSlider, RangeProps } from './RangeSlider';
