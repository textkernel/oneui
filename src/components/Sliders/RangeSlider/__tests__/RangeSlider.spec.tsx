import React from 'react';
import toJson from 'enzyme-to-json';
import { RangeSlider } from '../RangeSlider';

describe('RangeSlider component', () => {
    it('should render component correctly', () => {
        const wrapper = mount(<RangeSlider />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
