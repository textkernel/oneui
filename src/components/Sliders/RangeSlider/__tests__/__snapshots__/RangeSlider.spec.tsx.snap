// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeSlider component should render component correctly 1`] = `
<RangeSlider
  defaultValue={
    Array [
      0,
      100,
    ]
  }
>
  <ComponentEnhancer(Range)
    activeDotStyle={Object {}}
    allowCross={true}
    ariaLabelGroupForHandles={Array []}
    ariaLabelledByGroupForHandles={Array []}
    ariaValueTextFormatterGroupForHandles={Array []}
    className=""
    count={1}
    defaultValue={
      Array [
        0,
        100,
      ]
    }
    disabled={false}
    dotStyle={Object {}}
    dots={false}
    draggableTrack={false}
    handle={[Function]}
    handleStyle={
      Array [
        Object {},
      ]
    }
    included={true}
    marks={Object {}}
    max={100}
    min={0}
    onAfterChange={[Function]}
    onBeforeChange={[Function]}
    onChange={[Function]}
    prefixCls="rc-slider"
    pushable={false}
    railStyle={Object {}}
    reverse={false}
    step={1}
    tabIndex={Array []}
    trackStyle={
      Array [
        Object {},
      ]
    }
    vertical={false}
  >
    <div
      className="rc-slider"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="rc-slider-rail"
        style={Object {}}
      />
      <Track
        className="rc-slider-track rc-slider-track-1"
        included={true}
        key="1"
        length={100}
        offset={0}
        reverse={false}
        style={Object {}}
        vertical={false}
      >
        <div
          className="rc-slider-track rc-slider-track-1"
          style={
            Object {
              "left": "0%",
              "right": "auto",
              "width": "100%",
            }
          }
        />
      </Track>
      <Steps
        activeDotStyle={Object {}}
        dotStyle={Object {}}
        dots={false}
        included={true}
        lowerBound={0}
        marks={Object {}}
        max={100}
        min={0}
        prefixCls="rc-slider"
        reverse={false}
        step={1}
        upperBound={100}
        vertical={false}
      >
        <div
          className="rc-slider-step"
        />
      </Steps>
      <Handle
        className="rc-slider-handle rc-slider-handle-1"
        disabled={false}
        key="0"
        max={100}
        min={0}
        offset={0}
        prefixCls="rc-slider"
        reverse={false}
        style={Object {}}
        tabIndex={0}
        value={0}
        vertical={false}
      >
        <div
          aria-disabled={false}
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={0}
          className="rc-slider-handle rc-slider-handle-1"
          onBlur={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="slider"
          style={
            Object {
              "left": "0%",
              "right": "auto",
              "transform": "translateX(-50%)",
            }
          }
          tabIndex={0}
        />
      </Handle>
      <Handle
        className="rc-slider-handle rc-slider-handle-2"
        disabled={false}
        key="1"
        max={100}
        min={0}
        offset={100}
        prefixCls="rc-slider"
        reverse={false}
        tabIndex={0}
        value={100}
        vertical={false}
      >
        <div
          aria-disabled={false}
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={100}
          className="rc-slider-handle rc-slider-handle-2"
          onBlur={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="slider"
          style={
            Object {
              "left": "100%",
              "right": "auto",
              "transform": "translateX(-50%)",
            }
          }
          tabIndex={0}
        />
      </Handle>
      <Marks
        className="rc-slider-mark"
        included={true}
        lowerBound={0}
        marks={Object {}}
        max={100}
        min={0}
        onClickLabel={[Function]}
        reverse={false}
        upperBound={100}
        vertical={false}
      >
        <div
          className="rc-slider-mark"
        />
      </Marks>
    </div>
  </ComponentEnhancer(Range)>
</RangeSlider>
`;
